### PR TITLE
Improve gen_reflection_proto.sh

### PR DESCRIPTION
**Pull request by y-zeng**
- Add comments in generated files to state how to re-generate them.
- Check available protoc and gprc_cpp_plugin

### CODE REVIEW

File: include/grpc++/ext/reflection.grpc.pb.h [hunk 1]
```diff
@@ -32,7 +32,7 @@
  */
 
 
-// Generated by the gRPC protobuf plugin.
+// Generated by tools/codegen/extensions/gen_refelction_proto.sh
```
Comment (stanley-cheung) [inline include/grpc++/ext/reflection.grpc.pb.h]: Typo, in 4 places

File: tools/codegen/extensions/gen_reflection_proto.sh [hunk 2]
```diff
@@ -56,6 +75,9 @@ sed -i "s/\"${PROTO_FILE}.pb.h\"/<${INCLUDE_DIR/\//\\\/}\/${PROTO_FILE}.pb.h>/g"
 sed -i "s/\"${PROTO_FILE}.pb.h\"/<${INCLUDE_DIR/\//\\\/}\/${PROTO_FILE}.pb.h>/g" ${TMP_DIR}/${PROTO_FILE}.grpc.pb.cc
 sed -i "s/\"${PROTO_FILE}.grpc.pb.h\"/<${INCLUDE_DIR/\//\\\/}\/${PROTO_FILE}.grpc.pb.h>/g" ${TMP_DIR}/${PROTO_FILE}.grpc.pb.cc
 
+sed -i "1s/.*/\/\/ Generated by tools\/codegen\/extensions\/gen_refelction_proto.sh/g" ${TMP_DIR}/*.pb.h
```
Comment (stanley-cheung) [inline tools/codegen/extensions/gen_reflection_proto.sh]: Oh, I guess the source of the typo is here.

File: include/grpc++/ext/reflection.grpc.pb.h (hunk 1 above)
Comment (y-zeng) [inline include/grpc++/ext/reflection.grpc.pb.h]: Fixed. Thanks for the review!

File: tools/codegen/extensions/gen_reflection_proto.sh (hunk 2 above)
Comment (y-zeng) [inline tools/codegen/extensions/gen_reflection_proto.sh]: Fixed. :sweat_smile:

**Comment by stanley-cheung** [comment]
LGTM

**Comment by y-zeng** [comment]
Test failures seem irrelevant. Only sanity tests are affected.

**Pull request merged**